kernel/src/main/scala/me/rexim/morganey/ast/LambdaTerm.scala

```diff
@@ -99,12 +84,22 @@ case class LambdaApp(leftTerm: LambdaTerm, rightTerm: LambdaTerm) extends Lambda
 
   override val freeVars: Set[String] = leftTerm.freeVars ++ rightTerm.freeVars
 
+  private[ast] def nestedApplications(): (LambdaTerm, List[LambdaTerm]) = {
+    @tailrec
+    def go(app: LambdaApp, acc: List[LambdaTerm]): (LambdaTerm, List[LambdaTerm]) =
+      app match {
+        case LambdaApp(l: LambdaApp, r) => go(l, r :: acc)
```

Comment:
It is member of LambdaApp, so I guess it's the same situation as https://github.com/rexim/Morganey/pull/167#discussion_r77370535